### PR TITLE
keyring: Clean up secp256k1 signing with other small changes

### PR DESCRIPTION
**Pull request by kevinji**
These changes prep a larger PR for #2851.

## Description

The KeyRing has a few cleanup changes made. In particular, the signing logic has been updated:

- The `k256` library is no longer used; instead, `secp256k1` is used for both Evmos (aka Ethermint) and Cosmos.
- `.sign_ecdsa` is used for both Evmos and Cosmos as in the current implementation for Evmos, the recovery value `v` is just thrown out via `let (_, sig_bytes)`. I'm not super familiar with Evmos but this seems wrong at a glance, so I left a TODO to figure out how to fix this in the future.

This is the most invasive part of the change, so any help with suggesting how we could test that signing works the same before and after would be helpful.

Separately, there are some miscellaneous changes:

- Use `Secp256k1::signing_only()` instead of `Secp256k1::new()` for clarity.
- Move `sign_message` into `KeyEntry` and remove the duplicated functions in `KeyRing` and at the top level.
- Move `from_seed_file` into `KeyEntry`.
- Simplify the `get_address` implementation.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [x] Reviewed `Files changed` in the GitHub PR explorer.
- [x] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer/src/keyring.rs [hunk 1]
```diff
@@ -98,6 +99,40 @@ impl KeyEntry {
...
+                keccak256_hash(message).into()
+            }
+            AddressType::Cosmos | AddressType::Ethermint { .. } => Sha256::digest(message),
+        };
+
+        // SAFETY: hashed_message is 32 bytes, as expected in `Message::from_slice`,
+        // so `unwrap` is safe.
+        let message = Message::from_slice(&hashed_message).unwrap();
+
+        // TODO: Call `.sign_ecdsa_recoverable` for Evmos.
```
Comment (romac) [inline crates/relayer/src/keyring.rs:130]: Let's address this TODO before merging

Comment (romac) [reply crates/relayer/src/keyring.rs:130]: @tomtau We would love to hear what you think we should do here?

Comment (tomtau) [reply crates/relayer/src/keyring.rs:130]: it should be a recoverable signature, see the previous code:
https://github.com/informalsystems/hermes/pull/2863/files#diff-08bcf6496eddb7428a6a3c7894c2ac3e33be5b3767c0a27a1953eda3dbbabe96L391

you need to enable this feature on secp256k1: https://github.com/rust-bitcoin/rust-secp256k1/blob/master/Cargo.toml#LL28

Comment (tomtau) [reply crates/relayer/src/keyring.rs:130]: if the pubkey ends with `.ethsecp256k1.PubKey`

Comment (kevinji) [reply crates/relayer/src/keyring.rs:130]: Right I agree the old code calls `sign_ecdsa_recoverable` but the previous line `let (_, sig_bytes) =` throws away the recovery parameter, which is the first element of the tuple.

I [wrote some test code](https://github.com/kevinji/secp256k1-test/blob/main/src/lib.rs) to validate that the remaining bytes of `sign_ecdsa_recoverable` is the same as the bytes from `sign_ecdsa`.

This is why I was asking what should be done for Evmos as I believe the current behavior is already incorrect.

Comment (tomtau) [reply crates/relayer/src/keyring.rs:130]: Hm, indeed. Unlike Ethereum, Ethermint doesn't seem to recover pubkeys from signatures (but just looks them up for the account), so it should be ok to use `sign_ecdsa`.

Comment (yihuang) [reply crates/relayer/src/keyring.rs:130]: yeah, ethermint only do recovery for eth tx that's signed by ethereum compatible wallets, but ibc txs go through normal cosmos-sdk ante handler pipelines.

Comment (kevinji) [reply crates/relayer/src/keyring.rs]: Great, I removed the TODO per the conversation here and updated the comment a bit.

**Review by romac — APPROVED** [review]

**Pull request merged**